tests/explicit-nulls/neg/flow8.scala

```diff
@@ -0,0 +1,17 @@
+class Foo {
+
+  def foo(): Unit = {
+    val x: String|Null = ???
+    if (x == null) return ()
+    implicit val y: Int = x.length
+  }
+
+  // This case is not valid but foo() above is valid, because
+  // non-lazy value definitions exist between forward references
+  def fr(): Unit = {
+    val z = implicitly[Int] // error: forward reference is not allowed here
+    val x: String|Null = ???
+    if (x == null) return ()
+    implicit val y: Int = x.length
```

Comment:
Shouldn't we see an error here, since refchecks happen _after_ typing? I would expect to see a typing error here.